src/predicate.rs

```diff
@@ -0,0 +1,112 @@
+//! Predicate representations with required data to be executed during VM runtime
+
+use crate::interpreter::MemoryRange;
+
+use fuel_asm::Word;
+use fuel_tx::{ConsensusParameters, Transaction};
+
+use std::borrow::Borrow;
+
+/// Runtime representation of a predicate
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct RuntimePredicate {
+    program: MemoryRange,
+    idx: usize,
+}
+
+impl RuntimePredicate {
+    /// Memory slice with the program representation of the predicate
+    pub const fn program(&self) -> &MemoryRange {
+        &self.program
+    }
+
+    /// Index of the transaction input that maps to this predicate
+    pub const fn idx(&self) -> usize {
+        self.idx
+    }
+
+    /// Create a new runtime predicate from a transaction, given the input index
+    ///
+    /// Return `None` if the tx input doesn't map to an input coin with a predicate
+    pub fn from_tx<T>(params: &ConsensusParameters, tx: T, idx: usize) -> Option<Self>
+    where
+        T: Borrow<Transaction>,
+    {
+        tx.borrow()
+            .input_coin_predicate_offset(idx)
```

Comment:
Hmm it seems like we are missing support for `MessagePredicate` verification as well. Will this require changes to fuel-tx?

Reply:
created an issue here: https://github.com/FuelLabs/fuel-tx/issues/166

Reply:
also filed an issue here in fuel-vm, so we can add message predicate verification as a followup: https://github.com/FuelLabs/fuel-vm/issues/175